src/formats/tana/tana-import.ts

```diff
@@ -0,0 +1,306 @@
+				const associatedNode = this.nodes.get(id as string);
+				if (associatedNode) {
+					this.markSeen(associatedNode);
+				}
+			}
+		}
+	}
+
+	private convertMetaNode(node: TanaDoc | undefined, fragments: string[], indent: number): any {
+		const result: any = {};
+		if (!node) return;
+		this.markSeen(node);
+		const props = this.collectNodeProperties(node);
+		for (const [id, , v] of props) {
+			if (id == 'SYS_A13') {
+				result.tag = v;
+			}
+			else if (id == 'SYS_A55') {
+				result.checkbox = true;
+			}
+		}
+		return result;
+	}
+
+	private generateLink(id: string): string {
+		const tlNode = this.topLevelNodes.get(id);
+		if (tlNode) {
+			return '[[' + tlNode[1] + ']]';
+		}
+		const targetNode = this.nodes.get(id);
+		if (targetNode) {
+			if (targetNode.props._docType == 'url') {
+				this.markSeen(targetNode);
+				return targetNode.props.name;
+			}
+			const tlParent = this.findTopLevelParent(targetNode);
+			if (tlParent) {
+				const tlFileName = this.topLevelNodes.get(tlParent.id)![1];
+				return '[[' + tlFileName + '#^' + id.replace('_', '-') + ']]';
+			}
+		}
+
+		return '[[#]]';
+	}
+
+	private findTopLevelParent(node: TanaDoc): TanaDoc | null {
+		const ownerId = node.props._ownerId;
+		if (!ownerId) return null;
+		const ownerNode = this.nodes.get(ownerId);
+		if (ownerNode) {
+			if (this.topLevelNodes.has(ownerNode.id)) {
+				return ownerNode;
+			}
+			return this.findTopLevelParent(ownerNode);
+		}
+		return null;
+	}
+
+	private convertMarkup(text: string): string {
+		return text
+			.replace(inlineRefRegex, (_, id) => this.generateLink(id))
+			.replace(boldRegex, (_, content) => '**' + content + '**')
+			.replace(italicRegex, (_, content) => '*' + content + '*')
+			.replace(strikeRegex, (_, content) => '~~' + content + '~~')
+			.replace(codeRegex, (_, content) => '`' + content + '`');
+	}
+
+	private markSeen(node: TanaDoc) {
+		if (this.convertedNodes.has(node.id)) return;
+		this.convertedNodes.add(node.id);
+		this.enumerateChildren(node, (child) => this.markSeen(child));
+		if (node.props._metaNodeId) {
+			const metaNode = this.nodes.get(node.props._metaNodeId);
+			if (metaNode) {
+				this.markSeen(metaNode);
+			}
+		}
+		this.markAssociatedNodesSeen(node);
+	}
+
+	private enumerateChildren(node: TanaDoc, callback: (child: TanaDoc) => void) {
+		if (!node.children) return;
+		for (const childId of node.children) {
+			if (childId.startsWith('SYS_')) continue;
+			const childNode = this.nodes.get(childId);
+			if (childNode) {
+				callback(childNode);
+			}
+			else {
+				this.notices.push('Node with id ' + childId + ' (parent ' + (node.props.name ?? node.id) + ') not found');
+			}
+		}
+	}
+
+	private pathFromRoot(node: TanaDoc, root: TanaDoc): string | null {
+		if (node.props._ownerId) {
+			const owner = this.nodes.get(node.props._ownerId);
+			if (owner) {
+				let pathFromRoot = owner == root ? 'root' : this.pathFromRoot(owner, root);
+				if (pathFromRoot == null) return null;
```

Comment:
You can remove the `return null` since that is done below.

```
if (pathFromRoot !== null)
    return `${pathFromRoot} > ${node.props.name} [${node.id}]`;
```